### PR TITLE
feat: show PR activity per run in operator UI

### LINKED ISSUE #115: [Feature] Show PR activity (created/updated/closed) per run in UI
## Feature Request

It would be very useful to see what PRs each Renovate run created, updated, or closed — both as a count summary and as a list of PR links.

Currently the UI shows per-project run status (completed/failed) but no information about what the run actually *did*. To check if Renovate created or updated PRs, you have to dig into executor pod logs.

## What Renovate Already Logs

Renovate emits rich PR activity in its JSON logs that the operator could parse:

**New PR created:**
```json
{"level":30, "msg":"Creating PR", "branch":"renovate/some-dep", "title":"Update X to Y", ...}
```

**Existing PR updated or unchanged:**
```json
{"level":20, "msg":"Pull Request #101 does not need updating", "branch":"renovate/rook-packages", ...}
```

**Git push with PR link:**
```json
{"level":20, "msg":"git push", "branch":"renovate/rook-packages", "result":{"remoteMessages":{"all":["Visit the existing pull request:","https://git.wxs.ro/k8s/flux/pulls/101 merges into main"]}}, ...}
```

## Suggested Implementation

Extend the existing log parser (`logParser.go`) to also extract PR activity:

1. **Count PRs created/updated/unchanged per run** — parse "Creating PR", "Updating PR", "does not need updating" messages
2. **Store PR summary on the project status** — e.g. `PRsCreated: 2, PRsUpdated: 1, PRsUnchanged: 3`
3. **Show in UI** — a small badge or column showing PR counts per project run, with optional expansion to show PR titles/links

This would make the operator UI much more useful for day-to-day monitoring — at a glance you could see which repos had actual dependency updates vs. just a clean scan.

## Relationship to #35

This is complementary to the metrics work in #35. Metrics give Prometheus counters for alerting, while this feature provides human-readable PR details in the UI.

### PR DESCRIPTION
**Pull request by vtmocanu**
## Summary

Extract PR activity (created/updated/unchanged/automerged) from Renovate JSON logs and display it in the operator dashboard with expandable per-PR details.

- **PR Activity column** with labeled badges (purple=automerged, green=created, amber=updated, gray=unchanged)
- **Expandable accordion rows** showing per-PR details with clickable links to the forge
- **Complete branch coverage** via "branches info extended" summary backfill (includes branches skipped in the run)
- **Title backfill** from branches info for PRs captured via per-message parsing
- **URL backfill** from known URL prefix pattern (no forge API calls needed)
- **Stale branch filtering** (`result="already-existed"` excluded)
- **Multi-forge support** tested with GitHub, Forgejo, and GitLab URL formats
- **25 new test cases** covering edge cases, truncated logs, automerge, stale branch filtering

### Backend changes

- Extend log parser to detect 7 message types: "Creating PR", "Updating PR", "does not need updating", "git push", "PR created", "PR automerged", and "branches info extended"
- Add `PRAction`, `PRDetail`, `PRActivity` types to CRD API with `Automerged` count
- Add `PRActivity` to `RenovateStatusUpdate` DTO, applied via `updatePRActivity()` in `projectStatus.go`
- Fix `DeepCopyObject`/`DeepCopyInto` to properly deep-copy pointer and slice fields (bonus bugfix for `RenovateJobList`)
- Cap at 100 PRDetails to prevent CRD bloat
- Sort: automerged > created > updated > unchanged

### UI changes

- PR Activity column with expandable accordion sub-rows
- Shared `prActionBadgeClass()` utility (no duplication)
- Sanitized HTML IDs for project names containing `/`
- Expanded state persists across 30s poll refresh (`useState Set`)
- Mobile card view with inline expandable PR list
- `aria-expanded`/`aria-controls`, `focus-visible` ring for accessibility
- Explicit dark mode variants on all badge colors

## Test plan

- [x] `go test ./...` passes (25 new parser tests + all existing)
- [x] `go build ./...` compiles
- [x] CRD YAML regenerated via `controller-gen`
- [x] Deployed to live cluster, validated end-to-end with real Renovate runs
- [x] All badge types verified: created, updated, unchanged, automerged
- [x] URL and title backfill verified
- [x] Stale branch filtering verified
- [x] Code reviewed by 4 independent (agentic 😄) reviewers 

Closes #115

### CODE REVIEW

File: src/internal/parser/logParser.go [hunk 1]
```diff
@@ -78,10 +149,189 @@ func ParseRenovateLogs(logs string) *LogParseResult {
...
+			}
+
+		case entry.Msg == "branches info extended":
+			// Complete list of all branches, including those skipped in this run.
+			// Backfills branches not seen in per-message parsing.
+			// Skip branches with result="already-existed" (stale branches with closed/merged PRs).
+			var bi branchesInfoEntry
+			if err := json.Unmarshal([]byte(line), &bi); err == nil {
+				for _, b := range bi.BranchesInfo {
+					if b.BranchName == "" || b.Result == "already-existed" {
```
Comment (lukashankeln) [inline src/internal/parser/logParser.go]: I think `b.Result == "not-scheduled"` should also be skipped, i got the following in one of my Repos:

``` 
"branchesInformation": [
  {
    "branchName": "renovate/lock-file-maintenance",
    "prNo": null,
    "prTitle": "build(deps): lock file maintenance",
    "result": "not-scheduled",
    "upgrades": []
   }
]
```

Comment (vtmocanu) [reply src/internal/parser/logParser.go]: Good catch! I've switched to a whitelist approach instead of a blacklist. Now only branches with `result` of `"done"`, `"automerged"`, or `""` (empty/not set) are included as new entries. Everything else (`"already-existed"`, `"not-scheduled"`, and any future unknown values) is excluded.

Title backfill for branches already captured by per-message parsing still works regardless of the result value (moved the existing-branch check before the whitelist filter).

Updated in the latest push.

**Comment by vtmocanu** [comment]
Regarding the LOG_LEVEL concern - I tested this extensively and here's what I found:

**Renovate's default log level is debug (level 20)**. The operator only sets `LOG_LEVEL=debug` explicitly when `executionOptions.Debug` is true (see `jobDefinitions.go:244`). When debug is off, no `LOG_LEVEL` is set, so Renovate uses its default, which outputs level 20+ messages. This means PR activity works out of the box without any configuration.

Messages we parse and their levels (verified from actual executor pod logs):
| Message | Level | Available with LOG_LEVEL=info? |
|---------|-------|-------------------------------|
| `"Creating PR"` | 20 (debug) | No |
| `"Updating PR"` | unverified | Unknown |
| `"does not need updating"` | 20 (debug) | No |
| `"git push"` | 20 (debug) | No |
| `"PR created"` | 30 (info) | Yes |
| `"PR automerged"` | 30 (info) | Yes |
| `"branches info extended"` | 20 (debug) | No |

If a user explicitly sets `LOG_LEVEL=info` via `extraEnv`, they'd lose the debug-level messages. In that case, only `"PR created"` and `"PR automerged"` would be captured, and URL/branch backfill wouldn't work.

I went down a rabbit hole trying to add a tooltip/info icon in the UI to warn about this, but ran into several issues:
- `executionOptions.Debug` isn't the right signal, since Renovate defaults to debug output even without it
- Showing the icon on every project in every job was noisy
- We can't distinguish "no PRs because LOG_LEVEL=info suppressed them" from "genuinely no PRs for this repo" since both produce the same zero-count result
- The debug setting is job-level, but the icon would need to be per-project

In the end, I kept the UI clean with no tooltip. The default behavior works without configuration, and users who explicitly set `LOG_LEVEL=info` are making a deliberate choice.

Happy to add a note in the docs/README about this if you think it's worth documenting.
